Content.Server/Spawners/EntitySystems/GenericSpawnerSystem.cs

```diff
@@ -0,0 +1,111 @@
+using System.Linq;
+using System.Numerics;
+using Content.Server.GameTicking;
+using Content.Server.Spawners.Components;
+using Content.Shared.GameTicking.Components;
+using Content.Shared.Random.Helpers;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Server.Spawners.EntitySystems;
+
+/// <summary>
+/// This handles spawner markers.
+/// </summary>
+public sealed class GenericSpawnerSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _robustRandom = default!;
+    [Dependency] private readonly GameTicker _ticker = default!;
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<GameRuleStartedEvent>(OnRuleStarted);
+        SubscribeLocalEvent<GenericSpawnerComponent, MapInitEvent>(OnSpawnMapInit);
+    }
+
+    private void OnSpawnMapInit(EntityUid uid, GenericSpawnerComponent component, MapInitEvent args)
+    {
+        TrySpawn(uid, component);
+        if (component.DeleteSpawnerAfterSpawn)
+            QueueDel(uid);
+    }
+
+    private void OnRuleStarted(ref GameRuleStartedEvent args)
+    {
+        var query = EntityQueryEnumerator<GenericSpawnerComponent>();
+        while (query.MoveNext(out var uid, out var spawner))
+        {
+            RuleStarted(uid, spawner, args);
+        }
+    }
+
+    public void RuleStarted(EntityUid uid, GenericSpawnerComponent component, GameRuleStartedEvent obj)
+    {
+        if (component.GameRules.Contains(obj.RuleId))
+            Spawn(uid, component);
+    }
+
+    private void TrySpawn(EntityUid uid, GenericSpawnerComponent component)
+    {
+        if (component.GameRules.Count == 0)
+        {
+            Spawn(uid, component);
+            return;
+        }
+
+        foreach (var rule in component.GameRules)
+        {
+            if (!_ticker.IsGameRuleActive(rule))
+                continue;
+            Spawn(uid, component);
+            return;
+        }
+    }
+
+    private void Spawn(EntityUid uid, GenericSpawnerComponent component)
+    {
+        if (Deleted(uid))
+            return;
+
+        if (component.EntityTables.Count == 0)
+            return;
+
+        foreach (var pair in component.EntityTables)
```

Comment:
You have the logic reversed a bit at the moment.

Currently you have it set up so that every table gets x many rolls.

When instead it should be, for each roll, it should randomly pick a table and then something from that table.

Reply:
This will end up spawning multiple items in the same location if the roll for each table succeeds, even if you set the roll to 1

Reply:
fair